web/src/compose_recipient.js

```diff
@@ -186,36 +198,42 @@ export function on_compose_select_recipient_update(new_value) {
         const $stream_header_colorblock = $(
             "#compose_recipient_selection_dropdown .stream_header_colorblock",
         );
-        stream_bar.decorate(new_value, $stream_header_colorblock);
-        if (message_type === "private") {
-            switch_message_type("stream");
-        }
+        const stream_name = compose_state.stream_name();
+        stream_bar.decorate(stream_name, $stream_header_colorblock);
+        switch_message_type("stream");
```

Comment:
I remember there being issues with calling this unconditionally around resetting focus in undesirable way motivated our having that `if` statement, but I've not actually reproduced any, so maybe it's fine. @evykassirer do you remember why we had that if statement to not change the message type if it was already correct?

Reply:
Or maybe it's the other way around, looking at the comment for `switch_message_type("private");` -- we want to do the focus logic unconditionally and skipping it doesn't do that?

Reply:
`update_placeholder_text();` is the function I need to call here regardless of `message_type`, for which `switch_message_type` seems like the right function to call.